lib/parser/source/tree_rewriter/action.rb

```diff
@@ -69,14 +77,22 @@ def place_in_hierarchy(action)
           extra_sibbling = if family[:parent]  # action should be a descendant of one of the children
             family[:parent][0].do_combine(action)
           elsif family[:child]                 # or it should become the parent of some of the children,
-            action.with(children: family[:child])
+            action.with(children: family[:child], enforcer: @enforcer)
+              .combine_children(action.children)
           else                                 # or else it should become an additional child
             action
           end
           with(children: [*family[:sibbling], extra_sibbling])
         end
       end
 
+      # Assumes more_children all contained within @range
+      def combine_children(more_children)
+        more_children.inject(self) do |mother, new_child|
```

Comment:
let's rename `mother` to `parent` if it makes sense. It took me a few minutes to understand this code because of the unusual naming.

Reply:
Absolutely.